core/src/main/java/de/danoeh/antennapod/core/receiver/MediaButtonReceiver.java

```diff
@@ -30,7 +29,7 @@ public void onReceive(Context context, Intent intent) {
 			Intent serviceIntent = new Intent(context, PlaybackService.class);
 			serviceIntent.putExtra(EXTRA_KEYCODE, event.getKeyCode());
 			serviceIntent.putExtra(EXTRA_SOURCE, event.getSource());
-			ContextCompat.startForegroundService(context, serviceIntent);
+			context.startService(serviceIntent); // the service itself will determine if it needs to become a foreground service.
```

Comment:
This sometimes leads to `IllegalStateException`s. That's why I switched to startforegroundservice recently

> The startService() method now throws an IllegalStateException if an app targeting Android 8.0 tries to use that method in a situation when it isn't permitted to create background services.
> https://developer.android.com/about/versions/oreo/android-8.0-changes#atap

Reply:
Thanks for spotting it - I somehow missed it during analyzing logcat, mistakenly believing that something went wrong on the widget side.